src/modules/snobal/snobal.cpp

```diff
@@ -398,7 +397,23 @@ void snobal::run(mesh_elem &face)
 //        BOOST_THROW_EXCEPTION(module_error() << errstr_info ("Snobal died. Triangle center = "+details));
     }
 
-
+    // something has gone very wrong with, likely, a thin snowcover
+    if( std::isnan(sbal->m_s))
+    {
+        // "melt" what we had
```

Comment:
If I understand well, we do not keep track of the previous SWE values before the NaN occurred so that we are not really "melting" what we had. It is simply diseapperaing and nothing is added to the runoff. Am I correct?